packages/core/src/agent/didcomm/versions/v2/DIDCommV2Message.ts

```diff
@@ -0,0 +1,89 @@
+import type { ServiceDecorator } from '../../../../decorators/service/ServiceDecorator'
+import type { AgentMessage } from '../../../AgentMessage'
+
+import { parseUrl } from 'query-string'
+
+import { AriesFrameworkError } from '../../../../error'
+import { JsonEncoder } from '../../../../utils'
+import { JsonTransformer } from '../../../../utils/JsonTransformer'
+import { DIDCommMessageVersion } from '../../types'
+
+import { DIDCommV2BaseMessage } from './DIDCommV2BaseMessage'
+
+export class DIDCommV2Message extends DIDCommV2BaseMessage implements AgentMessage {
+  public toJSON(): Record<string, unknown> {
+    return JsonTransformer.toJSON(this)
+  }
+
+  public get version(): DIDCommMessageVersion {
+    return DIDCommMessageVersion.V2
+  }
+
+  public get threadId(): string | undefined {
+    return this.thid
+  }
+
+  public get sender(): string | undefined {
+    return this.from
+  }
+
+  public serviceDecorator(): ServiceDecorator | undefined {
+    return undefined
+  }
+
+  public hasAnyReturnRoute() {
+    return false
+  }
+
+  public hasReturnRouting() {
+    return false
+  }
+
+  public setReturnRouting(): void {
+    return
+  }
```

Comment:
I think we should throw errors here? Is this not implemented yet?

DIDcomm v2 supports return routing through an extension right?